### ISSUE TITLE
BHoM_Adapter: refactoring-remove NextId and AssignId

### ISSUE THREAD

**Issue by alelom** [BHoM/BHoM_Adapter#115]
### Proposed actions
- Remove the `AssignId` method completely.
- Id assignment to be done always in the `Create `method.
- Remove the `NextId` method completely.

### Explanation
The `AssignId` and `NextId` methods enforce on the user an additional structure that may not be needed.

We propose a shift of the Id assignment to the Create method in all cases.
This leads to simplification of the NextId method that can be removed as well.

This also helps clarifying the responsibility of the Create action.

**Comment by alelom** [comment]
After chat with @adecler and @kThorsager :
- initally we thought to move the method in the Create() just to facilitate its implementation
- following discussion Loads, evaluate how to store the IDs in the Adapter itself to avoid issue risen after the DeepClone

**Comment by alelom** [comment]
Action needed for Toolkits:
- If you had Config.UseAdapterId set to `true`, AND you are implementing the `Create()` method (e.g. it's not empty):
   - The `NextId()` method is now deleted in the base Adapter. The logic of the Id assignment should now go directly in the Create() method. It is not called automatically anymore by the base Adapter. 
You can either:
      - a) Put the logic for the id assignment in the Create() method, before creating the objects.
      - b) Keep the existing logic. To do this, you need to:
           - b1) Keep the `NextId()` as is, but removing the `override` from its signature
           - b2) Appropriately call the `NextId()` at the start of your `Create()` method(s). 
This is needed because the `AssignId()` method, that was in the base Adapter and was responsible for calling `NextId()`, has been deleted. You need to place its logic in your toolkit just before the object creation. 
In short, *you need to put the following as the first bit of code in your `Create()` method override*:
        ```cs
        foreach (var item in objects)
        {
            if (!item.CustomData.ContainsKey(AdapterId))
                    item.CustomData[AdapterId] = NextId(item.GetType(), true);
        }
       ```

**Comment by alelom** [comment]
An issue is that it becomes difficult to live with this:
https://github.com/BHoM/BHoM_Adapter/blob/7b6e2465e0acf0894ecc97ae22eeca8ade627b81/BHoM_Adapter/HelperMethods/CRUD.cs#L83-L89
that should be either done in all cases (slow because you need to check if Ids have been assigned for each object) or altogether removed.

I need to experiment with the Adapter ID dictionary.

**Comment by alelom** [comment]
Strategy changed in #164
